### DISCUSSION TITLE
[QUESTION] How to cross-compile to use in an Android app?

### DISCUSSION THREAD

**Discussion by mehmetcy84** [awslabs/amazon-kinesis-video-streams-webrtc-sdk-c#1731] [Q&A]
I know there is a separate Kinesis Android SDK, but I need to build the Kinesis embedded C SDK to use the built libraries in an Android project.

In cross-compilation section of the sdk overview page, certain environment variables are mentioned to be set. 
* Is it possible to build the Kinesis C SDK along with its all dependencies contained in the repository, to produce libraries that can be added to Android Studio for Android app development?
* What would be the selection of accepted parameters for these variables, and specifically for Android?

> If you wish to cross-compile CC and CXX are respected when building the library and all its dependencies. You will also need to set BUILD_OPENSSL_PLATFORM, BUILD_LIBSRTP_HOST_PLATFORM and BUILD_LIBSRTP_DESTINATION_PLATFORM. See our codecov.io for an example of this. Every commit is cross compiled to ensure that it continues to work.

**Comment by ziku-du-05** [comment]
I also need to build this lib for android application usage. 

My requirement is I need to send stream from sdcard via webrtc. But using android sdk I failed to send media files, thus I presume this c sdk could solve the problem, sending media data file from sdcard to other peer via webrtc.

Am I right?

**Reply by steczol** [reply]
I don't think it's possible to build it for Android right now. Different ARM architectures are supported, but Android lacks support of `ifaddrs`, which is used by this library. It is also missing in Windows, and you can see in CMake and the source code itself that Windows platform is handled differently in this regards.
If you want to build it for Android you would have to implement what's `ifaddrs` is doing the Android way.
